mirror-url.go

```diff
@@ -32,6 +32,39 @@ type mirrorURLs struct {
 	Error         *probe.Error `json:"-"`
 }
 
+func (m mirrorURLs) WithError(err *probe.Error) mirrorURLs {
+	m.Error = err
+	return m
+}
+
+func (m mirrorURLs) Equal(o interface{}) bool {
+	n, ok := o.(mirrorURLs)
+	if !ok {
```

Comment:
wouldn't this be solved by not taking an input as 'interface ? - why would you type assert which a compiler can do this for you?